src/js/angular/core/services.js

```diff
@@ -79,20 +79,45 @@ function MenuItemsProvider() {
     let productInfo;
 
     const items = [];
+    let itemsWithMissedParent = [];
 
     this.addItem = function (item) {
         if (angular.isUndefined(item.parent)) {
             if (!exists(items, item.label)) {
                 items.push(cloneItem(item, []));
+                // A new parent is registered so we try to process items which parents were missed.
+                this.parentAdded();
             }
         } else {
-            const parentItem = findParent(items, item.parent);
-            if (!angular.isUndefined(parentItem)) {
-                if (!exists(parentItem.children, item.label)) {
-                    parentItem.children.push(cloneItem(item, item.children ? item.children : []));
-                }
+            if (!this.addItemToParent(item)) {
+                // Save item because parent may not processed yet. We will try later to process it again.
+                itemsWithMissedParent.push(item);
+            }
+        }
+    };
+
+    this.addItemToParent = function (item) {
+        const parentItem = findParent(items, item.parent);
+        if (!angular.isUndefined(parentItem)) {
+            if (!exists(parentItem.children, item.label)) {
+                parentItem.children.push(cloneItem(item, item.children ? item.children : []));
+                return true;
             }
         }
+        return false;
+    }
+
+    this.parentAdded = function () {
```

Comment:
Please change the name so it is more descriptive of what it does, not when it's called.

Reply:
done